### DISCUSSION TITLE
Errors when running API locally

### DISCUSSION THREAD

**Discussion by horolug** [open-meteo/open-meteo#327] [General]
First of all, huge thanks for the amazing work that you are doing.

I am trying to setup API locally on Ubuntu 22.04.2 LTS (Using prebuilt Ubuntu jammy packages)

Local server seems to start, it sends back response with proper date range, and null values 
Response example
`{"latitude":52.5,"longitude":13.400009,"generationtime_ms":0.24008750915527344,"utc_offset_seconds":0,"timezone":"GMT","timezone_abbreviation":"GMT","hourly_units":{"time":"iso8601","temperature_2m":"°C"},"hourly":{"time":["2021-01-01T00:00",...,"2021-01-02T23:00"],"temperature_2m":[null,...,ull]}}`

As i understand it is to be expected until some real data is not downloaded.

But I am getting errors when trying to fetch any data.

As example, error when I run 

`openmeteo-api benchmark`

`[ WARNING ] cannotOpenFile(filename: "/Volumes/2TB_1GBs/data/master-MRI_AGCM3_2_S/temperature_2m_max_0.om", errno: 2, error: "No such file or directory") (Vapor/Application.swift:107)
[ DEBUG ] Application shutting down (Vapor/Application.swift:136)
Swift/ErrorType.swift:200: Fatal error: Error raised at top level: SwiftPFor2D.SwiftPFor2DError.cannotOpenFile(filename: "/Volumes/2TB_1GBs/data/master-MRI_AGCM3_2_S/temperature_2m_max_0.om", errno: 2, error: "No such file or directory")
Current stack trace:
0                                       0x0000559428265820 <unavailable> + 31025184
1                                       0x0000559427f640ef <unavailable> + 27873519
2                                       0x0000559427f63f07 <unavailable> + 27873031
3                                       0x0000559427f62e7c <unavailable> + 27868796
4                                       0x0000559427fc1b76 <unavailable> + 28257142
5                                       0x00005594279adc60 openmeteo_api_main + 329
6    libc.so.6                          0x00007f01eda29d90 <unavailable> + 171408
7    libc.so.6                          0x00007f01eda29dc0 __libc_start_main + 128
8                                       0x0000559426a11460 _start + 37
Received signal 4. Backtrace:
0x559426ecfe02, Backtrace.(printBacktrace in _B82A8C0ED7C904841114FDF244F9E58E)(signal: Swift.Int32) -> () at /__w/open-meteo/open-meteo/.build/checkouts/swift-backtrace/Sources/Backtrace/Backtrace.swift:66
0x559426ed0077, closure #1 (Swift.Int32) -> () in static Backtrace.Backtrace.install(signals: Swift.Array<Swift.Int32>) -> () at /__w/open-meteo/open-meteo/.build/checkouts/swift-backtrace/Sources/Backtrace/Backtrace.swift:80
0x559426ed0077, @objc closure #1 (Swift.Int32) -> () in static Backtrace.Backtrace.install(signals: Swift.Array<Swift.Int32>) -> () at /__w/open-meteo/open-meteo/<compiler-generated>:79
0x7f01eda4251f at ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x559427f62e88
0x559427fc1b75
0x5594279adda8, openmeteo_api_main at /__w/open-meteo/open-meteo/<compiler-generated>:0
0x7f01eda29d8f, __libc_start_call_main at ../sysdeps/nptl/libc_start_call_main.h:58
0x7f01eda29e3f, __libc_start_main_impl at ../csu/libc-start.c:392
0x559426a11484
0xffffffffffffffff
Illegal instruction (core dumped)
`

errors when I run 

`openmeteo-api download-ecmwf --run 00 --only-variables temperature_2m`

`[ INFO ] Downloading domain ECMWF run '2023-05-10T00:00' (App/DownloadEcmwfCommand.swift:52)
[ INFO ] Downloading height and elevation data (App/DownloadEcmwfCommand.swift:82)
[ INFO ] Downloading file https://data.ecmwf.int/forecasts/20230510/00z/0p4-beta/oper/20230510000000-0h-oper-fc.index (App/Curl.swift:81)
[ INFO ] Downloading [ranged] file https://data.ecmwf.int/forecasts/20230510/00z/0p4-beta/oper/20230510000000-0h-oper-fc.grib2 (App/Curl.swift:81)
[ INFO ] Generating elevation file (App/DownloadEcmwfCommand.swift:111)
Received signal 4. Backtrace:
0x55cf53116e02, Backtrace.(printBacktrace in _B82A8C0ED7C904841114FDF244F9E58E)(signal: Swift.Int32) -> () at /__w/open-meteo/open-meteo/.build/checkouts/swift-backtrace/Sources/Backtrace/Backtrace.swift:66
0x55cf53117077, closure #1 (Swift.Int32) -> () in static Backtrace.Backtrace.install(signals: Swift.Array<Swift.Int32>) -> () at /__w/open-meteo/open-meteo/.build/checkouts/swift-backtrace/Sources/Backtrace/Backtrace.swift:80
0x55cf53117077, @objc closure #1 (Swift.Int32) -> () in static Backtrace.Backtrace.install(signals: Swift.Array<Swift.Int32>) -> () at /__w/open-meteo/open-meteo/<compiler-generated>:79
0x7f1fc104251f at ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x55cf53737c60, p4nzenc128v16 at Sources/CTurboPFor/src/vp4c.c:415
0x55cf53ab30d4, function signature specialization <Arg[0] = Exploded> of generic specialization <Foundation.FileHandle> of SwiftPFor2D.OmFileWriterState.write(Swift.ArraySlice<Swift.Float>) throws -> () at /__w/open-meteo/open-meteo/Sources/SwiftPFor2D/SwiftPFor2D.swift:198
0x55cf53aaf4e1, generic specialization <Foundation.FileHandle> of SwiftPFor2D.OmFileWriterState.write(Swift.ArraySlice<Swift.Float>) throws -> () at /__w/open-meteo/open-meteo/Sources/SwiftPFor2D/SwiftPFor2D.swift:0
0x55cf53aaf4e1, generic specialization <Foundation.FileHandle> of SwiftPFor2D.OmFileWriter.write<A where A: SwiftPFor2D.OmFileWriterBackend>(fn: A, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, supplyChunk: (Swift.Int) throws -> Swift.ArraySlice<Swift.Float>) throws -> () at /__w/open-meteo/open-meteo/Sources/SwiftPFor2D/SwiftPFor2D.swift:350
0x55cf53aafc60, closure #1 () throws -> () in SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, supplyChunk: (Swift.Int) throws -> Swift.ArraySlice<Swift.Float>) throws -> () at /__w/open-meteo/open-meteo/Sources/SwiftPFor2D/SwiftPFor2D.swift:365
0x55cf53aafc60, SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, supplyChunk: (Swift.Int) throws -> Swift.ArraySlice<Swift.Float>) throws -> () at /__w/open-meteo/open-meteo/Sources/SwiftPFor2D/SwiftPFor2D.swift:363
0x55cf52d86695, SwiftPFor2D.OmFileWriter.write(file: Swift.String, compressionType: SwiftPFor2D.CompressionType, scalefactor: Swift.Float, all: Swift.Array<Swift.Float>) throws -> () at /__w/open-meteo/open-meteo/<compiler-generated>:0
0x55cf52d86695, (6) suspend resume partial function for function signature specialization <Arg[4] = Dead> of App.DownloadEcmwfCommand.downloadEcmwfElevation(application: Vapor.Application, domain: App.EcmwfDomain, base: Swift.String, run: App.Timestamp) async throws -> () at /__w/open-meteo/open-meteo/Sources/App/Ecmwf/DownloadEcmwfCommand.swift:125
0x55cf5418734d
0x55cf54187a2b
0x55cf53c0e5f3
0x55cf53c0e3a1
0x55cf53c19e81
0x7f1fc1094b42, start_thread at ./nptl/pthread_create.c:442
0x7f1fc11269ff at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
0xffffffffffffffff
Illegal instruction (core dumped)`

Errors with similar call stack seem to appear when I am trying to launch API via Docker

Maybe I am missing some local packages, or I missed some configuration step?

Thanks in advance

**Comment by patrick-zippenfenig** [comment]
The error in benchmark is expected. I only use this endpoint to test some low level performance issues. It contains hard-coded file locations.

The second error is not expected. It looks like your CPU does not support vector instructions. At least CPU after Intel Skylake is required. If you run it in a VM make sure to set the CPU to "host" and pass all CPU feature to the VM

**Reply by horolug** [reply]
Thanks for quick reply
I will try your suggestion
